wolfcrypt/src/port/Espressif/esp_crt_bundle/esp_crt_bundle.c

```diff
@@ -0,0 +1,1517 @@
+    if (ret == WOLFSSL_SUCCESS) {
+        /* Attempt to validate the certificate again */
+        ret = wolfSSL_CertManagerVerifyBuffer(cm, der, derSz,
+            WOLFSSL_FILETYPE_ASN1);
+
+        if (ret == WOLFSSL_SUCCESS) {
+            ESP_LOGCBI(TAG, "Successfully validated cert: %s\n", subject->name);
+
+            /* If verification is successful then override error. */
+            preverify = 1;
+        }
+        else {
+            ESP_LOGE(TAG, "Failed to verify cert after loading new CA. "
+                          "err = %d", ret);
+        }
+    }
+    else {
+        ESP_LOGE(TAG, "Failed to load CA");
+    }
+
+    /* We don't free the issue and subject, as they are
+     * pointers to current store->current_cert values. */
+    return preverify;
+}
+
+/* Not a Best Practice, but in dev one can ignore cert date/time: */
+#if defined(WOLFSSL_DEBUG_CERT_BUNDLE) && defined(WOLFSSL_DEBUG_IGNORE_ASN_TIME)
+static CB_INLINE int wolfssl_ssl_conf_verify_cb_before_date(int preverify,
+                                                WOLFSSL_X509_STORE_CTX* store)
+{
+    if (store == NULL) {
+        ESP_LOGE(TAG, "wolfssl_ssl_conf_verify_cb_before_date store is null");
+        preverify = 0;
+    }
+    else if ((preverify == 0) && (store->error == ASN_BEFORE_DATE_E)) {
+        ESP_LOGW(TAG, "Overriding ASN_BEFORE_DATE_E!");
+        preverify = 1;
+    }
+
+    return preverify;
+}
+
+static CB_INLINE int wolfssl_ssl_conf_verify_cb_after_date(int preverify,
+                                                WOLFSSL_X509_STORE_CTX* store)
+{
+    if (store == NULL) {
+        ESP_LOGE(TAG, "wolfssl_ssl_conf_verify_cb_after_date store is null");
+        preverify = 0;
+    }
+    else if ((preverify == 0) && (store->error == ASN_AFTER_DATE_E)) {
+        ESP_LOGW(TAG, "Overriding ASN_AFTER_DATE_E!");
+        preverify = 1;
+    }
+
+    return preverify;
+}
+#endif /* WOLFSSL_DEBUG_CERT_BUNDLE && WOLFSSL_DEBUG_IGNORE_ASN_TIME */
+
+#ifdef CONFIG_WOLFSSL_DEBUG_CERT_BUNDLE
+void print_cert_subject_and_issuer(WOLFSSL_X509_STORE_CTX* store)
+{
+    int i;
+    int totalCerts;
+
+    if (store == NULL) {
+        ESP_LOGCBI(TAG, "store is NULL");
+        totalCerts = 0;
+    }
+    else {
+        totalCerts = store->totalCerts;
+    }
+
+    for (i = 0; i < totalCerts; i++) {
+        WOLFSSL_BUFFER_INFO buffer = store->certs[i];
+        WOLFSSL_X509* cert = wolfSSL_X509_d2i(NULL,
+                                            (const unsigned char*)buffer.buffer,
+                                              buffer.length);
+        if (cert == NULL) {
+            ESP_LOGCBI(TAG, "Failed to parse certificate at index %d\n", i);
+            continue;
+        }
+
+        WOLFSSL_X509_NAME* subject = wolfSSL_X509_get_subject_name(cert);
+        WOLFSSL_X509_NAME* issuer = wolfSSL_X509_get_issuer_name(cert);
+
+        if (subject != NULL && issuer != NULL) {
+            char subjectStr[256], issuerStr[256];
```

Comment:
Maybe `X509_MAX_SUBJECT_LEN + 1`?

Reply:
Yes, that was clearly my intention as with the others. Thanks for catching this one missed.